fastcrawler_ui/controllers/spider.py

```diff
@@ -0,0 +1,54 @@
+from fastapi import HTTPException, status
+from fastcrawler import Depends, FastCrawler
+from fastcrawler.schedule.schema import Task
+
+from fastcrawler_ui.repository.spiders import SpiderRepository, TaskSettings
+
+
+class SpiderController:
+    def __init__(self, spider_repo: SpiderRepository = Depends(SpiderRepository)):
+        self.spider_repo = spider_repo
+
+    async def get_tasks(self, crawler: FastCrawler):
+        results = await self.spider_repo.get_tasks(crawler)
+        return [result.model_dump() for result in results]
+
+    async def get_process_by_task(self, crawler: FastCrawler, task_name: str):
+        results = self.spider_repo.get_tasks_in_processes(crawler)
+        for process, task in results.items():
+            if task.name == task_name:
+                return process
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"Task {task_name!r} not found",
+        )
+
+    async def start_task_by_name(self, crawler: FastCrawler, task_name: str):
+        process = await self.get_process_by_task(
+            crawler=crawler,
+            task_name=task_name,
+        )
+        await process.start()
+
+    async def stop_task_by_name(self, crawler: FastCrawler, task_name: str):
+        task = await self.get_process_by_task(
+            crawler=crawler,
+            task_name=task_name,
+        )
+        await task.stop()
```

Comment:
You should not use Spider to disable the task
Use the Process to disable a crawling process
Go in process.py, where class Process is defined. there is stop and start method, which works great.

That logic is written to ensure data integrity, and stop the process even if it's in middle of crawling. The request batch has to be performed, then on next batch, the while condition fails and hence the crawler start.

Remember that if you stop a crawler, it is stopped forever!
and if you start a crawler, it's stopped in rocketry. Maybe you should use background process for start. but stop is responding fast.